x/gamm/pool-models/balancer/amm_test.go

```diff
@@ -186,3 +187,114 @@ func TestCalculateAmountOutAndIn_InverseRelationship_ZeroSwapFee(t *testing.T) {
 		})
 	}
 }
+
+func TestCalcSingleAssetInAndOut_InverseRelationship(t *testing.T) {
+	type testcase struct {
+		initialPoolOut   int64
+		initialPoolIn    int64
+		initialWeightOut int64
+		tokenOut         int64
+		initialWeightIn  int64
+	}
+
+	// For every test case in testcases, apply a swap fee in swapFeeCases.
+	testcases := []testcase{
+		{
+			initialPoolOut:   1_000_000_000_000,
+			tokenOut:         100,
+			initialWeightOut: 100,
+			initialWeightIn:  100,
+		},
+		{
+			initialPoolOut:   1_000_000_000_000,
+			tokenOut:         100,
+			initialWeightOut: 50,
+			initialWeightIn:  100,
+		},
+		{
+			initialPoolOut:   1_000_000_000_000,
+			tokenOut:         50,
+			initialWeightOut: 100,
+			initialWeightIn:  100,
+		},
+		{
+			initialPoolOut:   1_000_000_000_000,
+			tokenOut:         100,
+			initialWeightOut: 100,
+			initialWeightIn:  50,
+		},
+		{
+			initialPoolOut:   1_000_000,
+			tokenOut:         100,
+			initialWeightOut: 100,
+			initialWeightIn:  100,
+		},
+		{
+			initialPoolOut:   2_351_333,
+			tokenOut:         7,
+			initialWeightOut: 148,
+			initialWeightIn:  57,
+		},
+		{
+			initialPoolOut:   1_000,
+			tokenOut:         25,
+			initialWeightOut: 100,
+			initialWeightIn:  100,
+		},
+		// {
```

Comment:
This test is failing. Question to reviewers - is this test invalid due to the low pool balance compared to tokenOut or this is a bug?

I have a feeling that it's the former but wanted to double-check

Reply:
That sounds right, as long as this is off by one

Reply:
Created an issue to uncomment: https://github.com/osmosis-labs/osmosis/issues/1359